bindings/nodejs/lib/types/block/address.ts

```diff
@@ -118,12 +133,89 @@ class NftAddress extends Address {
     }
 }
 
+/**
+ * An implicit account creation address.
+ */
+class ImplicitAccountCreationAddress extends Address {
+    /**
+     * The Ed25519 address.
+     */
+    readonly address: Ed25519Address;
+    /**
+     * @param address An Ed25519 address.
+     */
+    constructor(address: Ed25519Address) {
+        super(AddressType.ImplicitAccountCreation);
+        this.address = address;
+    }
+
+    toString(): string {
+        return this.address.toString();
+    }
+}
+
+/**
+ * An address with restricted capabilities.
+ */
+class RestrictedAddress extends Address {
+    /**
+     * The inner address.
+     */
+    readonly address: Address;
+    /**
+     * The allowed capabilities bitflags.
+     */
+    private allowed_capabilities: HexEncodedString = '0x00';
```

Comment:
why not either use `Uint8Array` or a bitmask internally and only convert to string when necessary?

Reply:
The problem is that I think the serialization just takes the class as a dictionary and it doesn't have additional logic afaik

Reply:
well theres no dicts in js, but it just calls .toJSON on each object. We can implement a custom one and itll be picked up.
There is nothing for parsing this though, making it difficult to actually use 2 way 😆

Reply:
(so it say not worth the effort)